pkg/value/value.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2016 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package value
+
+import (
+	"math"
+)
+
+const (
+	NormalNaN uint64 = 0x7ff8000000000001 // A quiet NaN. This is also math.NaN().
+	StaleNaN  uint64 = 0x7ff4000000000000 // A signalling NaN, starting 01 to allow for expansion.
```

Comment:
Oh never mind me, I thought you were referring to the last two hex digits of the overall number here - just looked up how `NaN`s are actually encoded and get it now. I also didn't know that "signaling `NaN`s" are an official concept. TIL.

Maybe to help the uninitiated like me, amend the comment to "A signaling NaN, with the significand starting with 01 to allow for expansion.". I guess a link to https://en.wikipedia.org/wiki/NaN somewhere around here would not hurt either :)

Also, is this part of the article unproblematic for us:

> in NaNs generated by the PA-RISC and MIPS processors, the signaling/quiet bit is zero if the NaN is quiet, and non-zero if the NaN is signaling. Thus, on these processors, the bit represents an 'is_signaling' flag.

Go doesn't yet have a MIPS port, but apparently there's [something experimental in the works](https://golang.org/doc/go1.6#ports). I'm wondering if a quiet NaN as we define it could end up as a signalling NaN there and possibly cause the CPU to throw exceptions when doing ops on it?